include/linux/ipts-binary.h

```diff
@@ -81,38 +84,62 @@ typedef enum {
 
 typedef struct ipts_bin_io_header {
 	char str[10];
-	unsigned short type;
+	u16 type;
 } ipts_bin_io_header_t;
 
 typedef struct ipts_bin_res_list {
-	unsigned int num;
+	u32 num;
 	ipts_bin_res_t res[];
 } ipts_bin_res_list_t;
 
 typedef struct ipts_bin_patch {
-	unsigned int index;
-	unsigned int reserved1[2];
-	unsigned int alloc_offset;
-	unsigned int patch_offset;
-	unsigned int reserved2;
+	u32 index;
+	u32 reserved1[2];
+	u32 alloc_offset;
+	u32 patch_offset;
+	u32 reserved2;
 } ipts_bin_patch_t;
 
 typedef struct ipts_bin_patch_list {
-	unsigned int num;
+	u32 num;
 	ipts_bin_patch_t patch[];
 } ipts_bin_patch_list_t;
 
 typedef struct ipts_bin_guc_wq_info {
-	unsigned int batch_offset;
-	unsigned int size;
+	u32 batch_offset;
+	u32 size;
 	char data[];
 } ipts_bin_guc_wq_info_t;
 
 typedef struct ipts_bin_bufid_patch {
-	unsigned int imm_offset;
-	unsigned int mem_offset;
+	u32 imm_offset;
+	u32 mem_offset;
 } ipts_bin_bufid_patch_t;
 
+typedef enum {
+	IPTS_DATA_FILE_FLAG_NONE,
+	IPTS_DATA_FILE_FLAG_SHARE,
+	IPTS_DATA_FILE_FLAG_ALLOC_CONTIGUOUS,
+} ipts_bin_data_file_flags_t;
```

Comment:
If you want these to be bit flags, they should have explicit values.

Unless this is something hidden in the firmware, and if that's true, we should document that because... ow!

Reply:
Their values should be 0, 1, and 2, which would already make them a bitflag. I agree it would be neccessary if we ever add another element to it (which will never happen probably because those come from the firmware)

Reply:
Yeah I think those should nevertheless be explicit, especially to document that this is not some random enum that could be reordered but the values are fixed by the firmware.

Reply:
Updated, but force pushing confused github so it still shows the old code for me here. The full diff on the PR works though.